scriptmodules/supplementary/audiosettings.sh

```diff
@@ -159,3 +178,65 @@ EOF
     mv "$tmpfile" "$home/.asoundrc"
     chown "$user:$user" "$home/.asoundrc"
 }
+
+function _pulseaudio_audiosettings() {
+    local sudo_env=XDG_RUNTIME_DIR=/run/user/$SUDO_UID
+    local cmd=(dialog --backtitle "$__backtitle" --menu "Set audio output (PulseAudio)." 22 86 16)
+    local options=()
+    local sink_index
+    local sink_label
+
+    # Check if PulseAudio is running, otherwise 'pacmd' will not work
+    if ! sudo -u "$user" "$sudo_env" pacmd stat>/dev/null; then
+        printMsgs "dialog" "PulseAudio is enabled, but not running\nAudio settings cannot be set right now"
+        return
+    fi
+    while read sink_index sink_label; do
+        options+=("$sink_index" "$sink_label")
+    done < <(sudo -u "$user" "$sudo_env" pacmd list-sinks | grep -e index -e alsa.name | awk -F [:=] '/index/ { idx=$2; getline; print idx,gensub(/"|bcm2835\s+/,"","g", $2) }')
```

Comment:
This line could be maybe split / laid out differently for readabilithy- if you're using awk you could probably lose grep also ?

Reply:
Re-arranged it a bit and made it entirely in `awk`. It's more readable, but not shorter.